utils/prof_utils.py

```diff
@@ -0,0 +1,1150 @@
+        comp_minima=[]
+        for i, root in enumerate(comp_roots):
+            idx = int(root)
+            left = comp_dict[key][idx-1]
+            if left>comp_dict[key][idx]:
+                comp_minima.append(root)
+            else:
+                comp_maxima.append(root)
+        #Turn the root locations into locations on profile, not on component
+        for root in comp_minima:
+            abs_root = root + comp_idx[key][0]
+            minima.append(abs_root)
+        for root in comp_maxima:
+            abs_root = root + comp_idx[key][0]
+            maxima.append(abs_root)
+
+    ignore_idx = []
+    for i, mx in enumerate(maxima):
+        if max(profile[int(mx-1):int(mx+1)]) < ignore_threshold*max(profile):
+            ignore_idx.append(i)
+    for i in sorted(ignore_idx, reverse=True):
+        del maxima[i]
+
+    return minima, maxima
+
+def find_minima_maxima_gauss(popt, pcov, x_length):
+    """
+    Finds all roots of a gaussian function
+
+    Parameters:
+    -----------
+    popt: list
+        A list of length 3N where N is the number of gaussians. This list contains the parameters amp, mean, centre respectively
+    pcov: np.matrix
+        The covariance matric corresponding to the parameters from popt
+    x_length: int
+        The length of the list used to fit the gaussian
+
+    Returns:
+    --------
+    minima: list
+        A list of the minimum points of the fit
+    maxima: list
+        A list of the maximum points of the fit
+    """
+    #Create the derivative list and spline it to find roots
+    x = np.linspace(0, x_length-1, x_length)
+    dy = multi_gauss_ddx(x, *popt)
+    spline_dy = UnivariateSpline(x, dy, s=0)
+    roots = spline_dy.roots()
+
+    #Find which are max and min
+    maxima = []
+    minima = []
+    for root in roots:
+        idx = int(root + 0.5)
+        if dy[idx-1] > dy[idx]:
+            maxima.append(root)
+        else:
+            minima.append(root)
+
+    minima_e = find_x_err(minima, popt, pcov)
+    maxima_e = find_x_err(maxima, popt, pcov)
+
+    return minima, maxima, minima_e, maxima_e
+
+#---------------------------------------------------------------
+def find_x_err(x, popt, pcov):
+    """
+    Finds the error in the horizontal position of a gaussian fit at the point x.
+    Uses the equation sigma_x = sigma_y/d2ydx2 where:
+    sigma_x = error in x
+    d2ydx2 = second derivative of the gaussian function at point x
+    sigma_y = J*C*J_T
+    J = Jacobian evalutated at point x
+    C = covariance matrix of gaussian fit
+    J_T = transposed jacobian
+
+    Parameters:
+    -----------
+    x: list
+        A list of points to evaluate the error at
+    popt: list
+        The parameters used to describe the gaussian fit
+    pcov: numpy.matrix
+        The covariance matrix corresponding to popt
+
+    Returns:
+    --------
+    x_err: list
+        The error evaluated at each point, x
+    """
+    x_err = []
+    for i, point in enumerate(x):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'i'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)